### PR TITLE
v5.2 release

### PR DESCRIPTION
**Pull request by jordan-smith721**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-node/blob/master/REVIEWING.md)

Staging - https://docs-mongodbcom-staging.corp.mongodb.com/node/docsworker-xlarge/v5.2-release/

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?

### CODE REVIEW

File: source/whats-new.txt [hunk 1]
```diff
@@ -30,6 +32,20 @@ Learn what's new in:
...
+.. _version-5.2:
+
+What's New in 5.2
+-----------------
+
+New features of the 5.2 {+driver-short+} release include:
+
+- The driver now supports automatically obtaining Azure credentials when using
+  automatic Queryable Encryption
+- Minor bug fixes
```
Comment (rustagir) [inline source/whats-new.txt:44]: C: IMO this isn't necessary. Probably all of the releases have minor bug fixes

File: source/whats-new.txt [hunk 2]
```diff
@@ -30,6 +32,20 @@ Learn what's new in:
...
 
+.. _version-5.2:
+
+What's New in 5.2
+-----------------
+
+New features of the 5.2 {+driver-short+} release include:
+
+- The driver now supports automatically obtaining Azure credentials when using
+  automatic Queryable Encryption
```
Comment (rustagir) [inline source/whats-new.txt:43]: S: maybe we can also link to https://www.mongodb.com/docs/v6.0/core/queryable-encryption/tutorials/azure/azure-automatic here like in the release notes

Comment (rustagir) [reply source/whats-new.txt:43]: also end with period

File: source/whats-new.txt [hunk 3]
```diff
@@ -105,6 +121,18 @@ New features of the 5.0 {+driver-short+} release include:
...
     manage credentials from the environment.
 
+.. _version-4.15:
+
+What's New in 4.15
+------------------
+
+New features of the 4.15 {+driver-short+} release include:
+
+- Support for AWS IAM roles for service accounts
```
Comment (rustagir) [inline source/whats-new.txt:131]: period

**Review by rustagir — APPROVED** [review]
approved, with some non-blocking comments

File: source/whats-new.txt (hunk 2 above)
Comment (jordan-smith721) [reply source/whats-new.txt]: Looking through that page, it doesn't look like we've actually documented this change there, so I think we should leave the link off for now

**Pull request merged**